src/main/java/org/truffleruby/debug/RubySingleThreadedTestLanguage.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.debug;
+
+import com.oracle.truffle.api.CallTarget;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.TruffleLanguage;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.impl.DefaultCallTarget;
+import com.oracle.truffle.api.nodes.RootNode;
+
+@TruffleLanguage.Registration(
+        name = "RubySingleThreadedTestLanguage",
+        id = "ruby-single-threaded-test",
+        mimeType = "application/x-ruby-single-threaded-test",
+        version = "0",
+        internal = true,
+        interactive = false)
```

Comment:
This should be in src/test/java if possible, and not part of truffleruby.jar.
I think we don't want to have an extra TruffleLanguage in the main distribution, whose sole purpose is testing.

Reply:
I'm not sure how to achieve that so it's on the class path. Do you know how we could do that?

It's not a big problem to have it in the distribution - it's just a few bytes, and it may be useful in diagnosing problems from users. It's also hidden from users as an internal, non-interactive language.

Reply:
On JVM, we'd just need to add `mxbuild/dists/truffleruby-test.jar` on the classpath for that test.

Right, if it's fully hidden from users then it that removes most of my concern.
I think it might still appear when Truffle shows a list of mime types or so though.